### PR TITLE
Use component siblings custom hook

### LINKED ISSUE #565: Create a hook for working with siblings elements
In #563 we introduced an internal function to work with siblings in the core/Sidebar component. It works with the DOM API directly and it works. However, would be nice to extract it to a hook, which is a bit more _React alike_ and will allow (probably among other benefits):

* Better reusability
* Better testing
* A central place for implementing a logic to be polite and restore the previous attributes values if they were present instead of simply removing them.

### PR DESCRIPTION
**Pull request by balsa-asanovic**
## Problem

In the Sidebar component an internal function to access its siblings was introduced.
This pull request address issue #565 by creating a custom hook for this purpose and using it instead in Sidebar component.

fixes #565 

## Solution

The logic of getting HTML node's siblings is moved to a custom hook, where also the functions are introduced for adding and removing attributes to sibling elements. The hook returns these functions as an array. The hook takes an HTML element as input parameter in form of useRef.current.

## Testing

At the moment I haven't added any unit tests for the hook, before I get some feedback on the code changes I am not sure if the approach I made is good at all.

I tested the functionality in the app, this works, I can confirm that the inert and aria attributes are being added and removed appropriately.

## Screenshots

image

### CODE REVIEW

File: web/src/hooks/useNodeSiblings.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/**
+ * Returns two functions, one which adds given attributes to siblings
+ * of the node passed as the hook parameter, and another one which removes attributes.
+ *
+ * @param {HTMLElement} node
+ * @returns {Array}
+ */
```
Comment (dgdavid) [inline web/src/hooks/useNodeSiblings.js:7]: Just a suggestion, 

```suggestion
/**
 * Function for adding an attribute to a sibling
 * 
 * @typedef {function} addAttributeFn
 * @param {string} attribute - attribute name
 * @param {*} value - value to set
 */

/**
 * Function for removing an attribute from a sibling
 * 
 * @typedef {function} removeAttributeFn
 * @param {string} attribute - attribute name
 */
    
/**
 * A hook for working with siblings of the node passed as parameter
 *
 * It returns an array with exactly two functions:
 *   - First for adding given attribute to siblings
 *   - Second for removing given attributes from siblings
 *
 * @param {HTMLElement} node
 * @returns {[addAttributeFn, removeAttributeFn]}
 */
```

**Review by dgdavid — COMMENTED** [review]
First and foremost, thanks a ton for taking care of this :heart: 

We apologize for the delay in reviewing it; we've been busy with the changes for the June release.

That said, your changes looks in the right direction. I left a few comments, but feel free to just add the testing part and ping us again for merging it to master branch as soon as all checks are green. 

Do not hesitate to contact us if you have questions about unit testing or filling out the changes file.

**Comment by balsa-asanovic** [comment]
Hey @dgdavid 

Thanks for the review and for the suggestions.

I addressed them in the code and made commits accordingly.

I've also added unit tests now for the custom hook.

Do tell me if anything else is needed for the PR to get merged.
I've noticed you mentioned "filling out the changes file", but I am not sure what to do here, so any help is appreciated.

**Review by imobachgs — CHANGES_REQUESTED** [review]
Thanks @balsa-asanovic! The only missing piece is adding an entry to the [changes file](https://github.com/openSUSE/agama/blob/master/web/package/cockpit-agama.changes). I would suggest something like this (but feel free to use a different text):

```
-------------------------------------------------------------------
Wed Jun 21 05:27:53 UTC 2023 - Balsa Asanovic <balsaasanovic95@gmail.com>

- Move the logic to add/remove attributes to sibling elements
  to a custom hook (gh#openSUSE/agama#565).

```

**Comment by balsa-asanovic** [comment]
Hey @imobachgs 

Thanks for pointing me to the right file.

I've added short description of changes, I hope it's OK.

**Review by imobachgs — APPROVED** [review]

**Comment by imobachgs** [comment]
Thank you! It looks OK. I will merge the PRs as soon as the CI runs.

**Comment by imobachgs** [comment]
The code is finally merged. @balsa-asanovic Thanks a lot for your contribution!

**Comment by balsa-asanovic** [comment]
Great!

Thank you guys a lot for the help and guidance. @dgdavid @imobachgs 

I'll look now for another issue to work on 😃

**Pull request merged**